kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/BrokerData.java

```diff
@@ -0,0 +1,237 @@
+import java.util.Objects;
+import java.util.StringJoiner;
+import javax.annotation.Nullable;
+
+/**
+ * A KafkaBroker resource type.
+ */
+public final class BrokerData {
+
+  private final ResourceLink links;
+
+  private final Attributes attributes;
+
+  private final Relationships relationships;
+
+  public BrokerData(
+      ResourceLink links,
+      String clusterId,
+      Integer brokerId,
+      @Nullable String host,
+      @Nullable Integer port,
+      @Nullable String rack,
+      Relationship configurations,
+      Relationship partitionReplicas) {
+    this.links = Objects.requireNonNull(links);
+    attributes = new Attributes(clusterId, brokerId, host, port, rack);
+    relationships = new Relationships(configurations, partitionReplicas);
+  }
+
+  @JsonProperty("type")
+  public String getType() {
+    return "KafkaBroker";
+  }
+
+  @JsonProperty("id")
+  public String getId() {
+    return String.format("%s/%s", attributes.getClusterId(), attributes.getBrokerId());
+  }
+
+  @JsonProperty("links")
+  public ResourceLink getLinks() {
+    return links;
+  }
+
+  @JsonProperty("attributes")
+  public Attributes getAttributes() {
+    return attributes;
+  }
+
+  @JsonProperty("relationships")
+  public Relationships getRelationships() {
+    return relationships;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    BrokerData that = (BrokerData) o;
+    return Objects.equals(links, that.links)
+        && Objects.equals(attributes, that.attributes)
+        && Objects.equals(relationships, that.relationships);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(links, attributes, relationships);
+  }
+
+  @Override
+  public String toString() {
+    return new StringJoiner(", ", BrokerData.class.getSimpleName() + "[", "]")
+        .add("links=" + links)
+        .add("attributes=" + attributes)
+        .add("relationships=" + relationships)
+        .toString();
+  }
+
+  public static final class Attributes {
+
+    private final String clusterId;
+
+    private final Integer brokerId;
+
+    @Nullable
+    private final String host;
+
+    @Nullable
+    private final Integer port;
+
+    @Nullable
+    private final String rack;
+
+    public Attributes(
+        String clusterId,
+        Integer brokerId,
+        @Nullable String host,
+        @Nullable Integer port,
+        @Nullable String rack) {
+      this.clusterId = Objects.requireNonNull(clusterId);
+      this.brokerId = Objects.requireNonNull(brokerId);
+      this.host = host;
+      this.port = port;
+      this.rack = rack;
+    }
+
+    @JsonProperty("cluster_id")
+    public String getClusterId() {
+      return clusterId;
+    }
+
+    @JsonProperty("broker_id")
+    public Integer getBrokerId() {
+      return brokerId;
+    }
+
+    @JsonProperty("host")
```

Comment:
Perhaps I'm misunderstanding the intent of this class but should we be using 'listener_host' and 'listener_post' here ? The one -pager uses 9093 as an example port which is pretty typical for an advertised listener. Furthermore should this actually be an array as brokers may have multiple listeners which may be bound to different interfaces? And does it make sense to advertise the security protocol as well? i.e. SASL_PLAINTEXT, SASL_SSL etc. The sasl mechanism itself isn't necessary as that is negotiated by the client anyway. If the idea is to leave that up to the configuration resource then perhaps the port doesn't belong here at all.

Reply:
the components of the result are pretty much dictated by `DescribeClusterResult` and the result of that i'm guessing is dependent on the connection used to call it. but your point seems good - why not list listeners? seems better? maybe @cmccabe has input as the root of the decision is pretty much the describe cluster call. But given the APIs, what's here sgtm.

i think the word listener is great if encapsulating everything @rnpridgeon is talking about, but if there's not more than one, it's implicit.

Reply:
This is what is currently exposed by AdminClient, which I assume to be the advertised listener, which is what the clients use. I can't see a mechanism to find the security information from the AdminClient currently. This is definitely something we will have to improve on later. But for now, keeping as is, as it will feel familiar to whoever knows the AdminClient.